apifest-oauth20/src/main/java/com/apifest/oauth20/AuthorizationServer.java

```diff
@@ -102,6 +108,14 @@ public ClientCredentials issueClientCredentials(HttpRequest req) throws OAuthExc
         return creds;
     }
 
+    private boolean areClientCredentialsValid(String clientId, String clientSecret) {
+        Pattern pattern = Pattern.compile(HttpRequestHandler.CLIENT_CREDENTIALS_PATTERN_STRING);
```

Comment:
No need to compile the pattern every time. This is a heavy operation and could be done once.